SO-4285 Validate request changes

AAAlinaaa opened the pull request.

AAAlinaaa committed:
SO-4285: Add componentURI to validation issues during validate requests

AAAlinaaa committed:
SO-4285: Add new validation configuration constants

cmark commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/validation/ValidationConfiguration.java:
```diff
@@ -29,6 +29,8 @@
 	public static final String USE_FSN = "useFsn";
 	public static final String LOCALES = "extendedLocales";
+	public static final String CODE_SYSTEMS = "codeSystems";
+	public static final String CODE_SYSTEM = "codeSystem";
```
Why do you need both?

cmark commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/validation/ValidateRequest.java:
```diff
@@ -159,34 +155,29 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 										ruleId,
-										branchPath,
-										componentIdentifier,
+										ComponentURI.of(codeSystemURI, componentIdentifier.getTerminologyComponentId(), componentIdentifier.getComponentId()),
```
Let's have a factory method on `ComponentURI` that receives a `ComponentIdentifier`, so it's easier to convert one from the other. Add it to the other PR #707

cmark commented on snomed/com.b2international.snowowl.validation.snomed/src/test/java/com/b2international/snowowl/validation/snomed/GenericValidationRuleTest.java:
```diff
@@ -59,6 +60,24 @@
 public class GenericValidationRuleTest extends BaseGenericValidationRuleTest {
 	
+	@Test
+	public void codeSystemURITest() throws Exception {
```
This should be added to the core tests, it should not part of the SNOMED CT Validation Rule module.

AAAlinaaa committed:
SO-4285: Remove no longer needed validation configuration constants

AAAlinaaa committed:
SO-4285: Rename constructor parameter to indicate it accepts a URI

AAAlinaaa committed:
SO-4285: Add further componentURI parse tests

AAAlinaaa committed:
SO-4285: Add id argument check, id argument test

AAAlinaaa committed:
SO-4285: Add ComponentURI factory method that accepts ...

... ComponentIdentifiers

AAAlinaaa committed:
Merge branch 'issue/SO-4285-component_uri_constructor_changes' into issue/SO-4285-validate_request_changes

AAAlinaaa committed:
SO-4285: Use new ComponentURI constructor

AAAlinaaa committed:
SO-4285: Commit progress on switching from branchPath to CodeSystemURI

AAAlinaaa committed:
SO-4285: Use CodeSystemURIs in handling issue delete requests

AAAlinaaa committed:
SO-4285: Base unique validation job id creation on code system URIs

AAAlinaaa committed:
SO-4285: Update validations REST API to use CodeSystemURIs

cmark commented on core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/validation/RepositoryValidationRestService.java:
```diff
@@ -109,16 +110,16 @@ public RepositoryValidationRestService(String repositoryId) {
 				.setRuleIds(validationInput.getRuleIds())
-				.build(repositoryId, validationInput.getBranchPath())
-				.runAsJobWithRestart(uniqueJobId, String.format("Validating branch '%s'", validationInput.getBranchPath()))
+				.build(new CodeSystemURI(validationInput.getCodeSystemURI()))
```
No need to wrap into CodeSystemURI here.

cmark commented on core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/validation/ValidationRestInput.java:
```diff
@@ -25,19 +25,19 @@
 
 	@NotEmpty
-	private String branchPath;
+	private String codeSystemURI;
```
This changes the API in a non-backward compatible manner, please keep the `branchPath` argument and make it deprecated.
Introduce a new `resource` argument instead of the `codeSystemURI` and mark it as a replacement of the `branchPath`.
Read order should be, `resource` first, if that is empty then fall back to `branchPath`.

cmark commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java:
```diff
@@ -104,18 +104,29 @@ public final ComponentIdentifier toComponentIdentifier() {
 		checkArgument(terminologyComponentId >= TerminologyRegistry.UNSPECIFIED_NUMBER_SHORT,
 				"Terminology component id should be either unspecified (-1) or greater than zero. Got: '%s'.", terminologyComponentId);
-		this.codeSystemUri = codeSystem;
+		boolean isUnspecified = codeSystemURI.getCodeSystem().equals(TerminologyRegistry.UNSPECIFIED) 
```
Swap left and right of the equals check.

cmark committed:
Merge branch '7.x' into issue/SO-4285-validate_request_changes

Conflicts:
	core/com.b2international.snowowl.core.tests/src/com/b2international/snowowl/core/uri/ComponentURITest.java
	core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java

cmark committed:
SO-4285: make validation REST API backward compatible

cmark committed:
SO-4285: rename codeSystemURI to resourceURI

cmark committed:
SO-4285: fix failed SnomedQueryValidationRuleEvaluatorTest

cmark committed:
SO-4285: fix backward compatibility issues in ValidationIssue

Rename componentURI to affectedComponentURI.
Fix a few naming issues.
Reduce old ctor visibility.
